packages/postgress/src/MessagePickupRepositoryPg.ts

```diff
@@ -0,0 +1,616 @@
+        encryptedMessage: message.encryptedmessage,
+        state: !deleteMessages ? 'sending' : message.state,
+      }))
+
+      return queuedMessages
+    } catch (error) {
+      this.logger?.error(`[takeFromQueue] Error: ${error}`)
+      return []
+    }
+  }
+
+  /**
+   * Retrieves the count of available messages in the queue for a given connection.
+   *
+   * @param {GetAvailableMessageCountOptions} options - Options for retrieving the message count.
+   * @param {string} options.connectionId - The ID of the connection to check.
+   * @returns {Promise<number>} A promise resolving to the count of available messages.
+   */
+  public async getAvailableMessageCount(options: GetAvailableMessageCountOptions): Promise<number> {
+    const { connectionId } = options
+    this.logger?.debug(`[getAvailableMessageCount] Initializing method for ConnectionId: ${connectionId}`)
+
+    try {
+      // Query to count pending messages for the specified connection ID
+      const query = `
+      SELECT COUNT(*) AS count 
+      FROM queuedmessages 
+      WHERE connectionid = $1 AND state = 'pending'
+    `
+      const params = [connectionId]
+      const result = await this.messagesCollection?.query(query, params)
+
+      if (!result || result.rows.length === 0) {
+        this.logger?.debug(`[getAvailableMessageCount] No pending messages found for ConnectionId: ${connectionId}`)
+        return 0
+      }
+
+      // Parse the count result
+      const numberMessage = parseInt(result.rows[0].count, 10)
+      this.logger?.debug(`[getAvailableMessageCount] Count of available messages: ${numberMessage}`)
+
+      return numberMessage
+    } catch (error) {
+      this.logger?.error(`[getAvailableMessageCount] Error while retrieving message count: ${error}`)
+      return 0
+    }
+  }
+
+  /**
+   * Adds a new message to the queue and processes it based on live session status.
+   *
+   * @param {AddMessageOptions} options - The options for adding a message.
+   * @param {string} options.connectionId - The ID of the connection.
+   * @param {string[]} options.recipientDids - Recipient DIDs for the message.
+   * @param {string} options.payload - The encrypted message payload.
+   * @returns { messageId, receivedAt:Promise<string> }- A promise resolving to the messageId and receivedAt of the added message.
+   * @throws {Error} Throws an error if the agent is not defined or if an error occurs during message insertion or processing.
+   */
+  public async addMessage(options: AddMessageOptions): Promise<string> {
+    const { connectionId, recipientDids, payload } = options
+    this.logger?.debug(`[addMessage] Initializing new message for connectionId: ${connectionId}`)
+
+    // Ensure the agent is defined
+    if (!this.agent) {
+      throw new Error('Agent is not defined')
+    }
+
+    try {
+      // Retrieve live session details for the given connection ID
+      const liveSession = await this.getLocalliveSession(connectionId)
```

Comment:
```suggestion
      const localLiveSession = await this.getLocalliveSession(connectionId)
```

To be more explicit in our intent

Reply:
Solved!